test/test_ops.py

```diff
@@ -45,6 +48,89 @@ def test_unsupported_dtypes(self, device, dtype, op):
             sample = samples[0]
             op(sample.input, *sample.args, **sample.kwargs)
 
+    # Helper for comparing torch tensors and numpy arrays
+    def assertEqualHelper(self, actual, expected, msg, *, dtype, exact_dtype=True, **kwargs):
+        assert isinstance(actual, torch.Tensor)
+
+        # Some NumPy functions return scalars, not arrays
+        if isinstance(expected, Number):
+            self.assertEqual(actual.item(), expected, **kwargs)
+        elif isinstance(expected, np.ndarray):
+            # Handles exact dtype comparisons between arrays and tensors
+            if exact_dtype:
+                # Allows array dtype to be float32 when comparing with bfloat16 tensors
+                # since Numpy doesn't support the bfloat16 dtype
+                # Also ops like scipy.special.erf, scipy.special.erfc etc. promote float16
+                # to float32
+                if expected.dtype == np.float32:
+                    assert actual.dtype in (torch.float16, torch.bfloat16, torch.float32)
+                else:
+                    assert expected.dtype == torch_to_numpy_dtype_dict[actual.dtype]
+
+            self.assertEqual(actual,
+                             torch.from_numpy(expected).to(actual.dtype),
+                             msg,
+                             exact_device=False,
+                             **kwargs)
+        else:
+            self.assertEqual(actual, expected, msg, exact_device=False, **kwargs)
+
+    def _test_reference_numerics(self, dtype, op, tensors, equal_nan=True):
```

Comment:
Would be nice if the main logic of `_test_reference_numerics` and `assertEqualHelper` in this file and test_unary_ufunc is consolidated at a common place and then imported by these tests.

If we have to tweak anything in future, we'll have to do it in a single place.

Reply:
Oh, sorry. I missed adding this point in the TODO list, once migration is done and I'm done testing that it doesn't affect the Unary Ufuncs reference checks, then I'll bring them all to a single place. Thanks for pointing this out, @kshitij12345 !

Reply:
More generally, I wonder if we should think about adding a helper function in common_utils.py that better supports testing against a NumPy reference. We have some old stuff but nothing modern.